src/openssl/ssl/server.cr

```diff
@@ -0,0 +1,46 @@
+require "socket"
+
+class OpenSSL::SSL::Server
+  include ::Socket::Server
+
+  getter wrapped : ::Socket::Server
+  getter context : OpenSSL::SSL::Context::Server
+
+  # If `#sync_close?` is `true`, closing this server will
+  # close the wrapped server.
+  property? sync_close : Bool
+
+  getter? closed : Bool = false
+
+  def initialize(@wrapped : ::Socket::Server, @context : OpenSSL::SSL::Context::Server = OpenSSL::SSL::Context::Server.new, @sync_close : Bool = true)
+  end
+
+  def self.open(wrapped : ::Socket::Server, context : OpenSSL::SSL::Context::Server = OpenSSL::SSL::Context::Server.new, sync_close : Bool = true)
+    server = new(wrapped, context, sync_close)
+
+    begin
+      yield server
+    ensure
+      server.close
```

Comment:
Maybe we should bubble exceptions in general. There might be a specific kind of exception that make sense to not bubble, but usually that is not the case. Other than that LGTM.

Reply:
This does bubble exceptions, it's `ensure` not `rescue`?

Reply:
Yup, I miss read it.